src/test/encoding/check-generated.sh

```diff
@@ -6,93 +6,121 @@ source $CEPH_ROOT/qa/standalone/ceph-helpers.sh
 
 dir=$1
 
-tmp1=`mktemp /tmp/typ-XXXXXXXXX`
-tmp2=`mktemp /tmp/typ-XXXXXXXXX`
-tmp3=`mktemp /tmp/typ-XXXXXXXXX`
-tmp4=`mktemp /tmp/typ-XXXXXXXXX`
+test_select_type() {
+    local type=$1
+    local result_dir=$2
+    local failed=0
+    local numtests=0
+    local pids=""
+    local result=$(mktemp ${result_dir}/$type-result-XXXXXXXX)
+    local tmp1=$(mktemp /tmp/typ-XXXXXXXX)
+    local tmp2=$(mktemp /tmp/typ-XXXXXXXX)
+    local tmp3=$(mktemp /tmp/typ-XXXXXXXX)
+    local tmp4=$(mktemp /tmp/typ-XXXXXXXX)
+    local num=`ceph-dencoder type $type count_tests`
 
-failed=0
-numtests=0
+    echo "$num $type"
+    for n in $(seq 1 1 $num 2>/dev/null); do
+        run_in_background pids save_stdout "$tmp1" ceph-dencoder type "$type" select_test "$n" dump_json
+        run_in_background pids save_stdout "$tmp2" ceph-dencoder type "$type" select_test "$n" encode decode dump_json
+        run_in_background pids save_stdout "$tmp3" ceph-dencoder type "$type" select_test "$n" copy dump_json
+        run_in_background pids save_stdout "$tmp4" ceph-dencoder type "$type" select_test "$n" copy_ctor dump_json
+        wait_background pids
+
+        if [ $? -ne 0 ]; then
+            echo "**** $type test $n encode+decode check failed ****"
+            echo "   ceph-dencoder type $type select_test $n encode decode"
+            failed=$(($failed + 3))
+            continue
+        fi
+
+        # nondeterministic classes may dump nondeterministically.  compare
+        # the sorted json output.  this is a weaker test, but is better
+        # than nothing.
+        local deterministic=0
+        if ceph-dencoder type "$type" is_deterministic; then
+            deterministic=1
+        fi
```

Comment:
we can take that out of the number test loop, it will be the same for each test number. it will also save some time..

Reply:
done.